common/v2/services/Analytics/Analytics.ts

```diff
@@ -0,0 +1,55 @@
+import { AxiosInstance } from 'axios';
+
+import {
+  ANALYTICS_API_URL,
+  ANALYTICS_ID_SITE,
+  ANALYTICS_ID_DESKTOP,
+  ANALYTICS_REC
+} from './constants';
+import { APIService } from '../API';
+import { isDevelopment, isDesktop } from 'v2/utils';
+import { Params, CvarEntry } from './types';
+
+let instantiated: boolean = false;
+export default class AnalyticsService {
+  public static instance = new AnalyticsService();
+
+  private service: AxiosInstance = APIService.generateInstance({
+    baseURL: ANALYTICS_API_URL
+  });
+
+  constructor() {
+    if (instantiated) {
+      throw new Error(`AnalyticsService has already been instantiated.`);
+    } else {
+      instantiated = true;
+    }
+  }
+
+  public trackPageVisit(pageUrl: string): Promise<any> {
+    const customParams: Params = {
+      local: isDevelopment().toString()
+    };
+
+    const cvar: object = this.mapParamsToCvars(customParams);
+
+    const analyticsId = isDesktop() ? ANALYTICS_ID_DESKTOP : ANALYTICS_ID_SITE;
+
+    const params: object = {
```

Comment:
Should we add a `timeout` here so we don't run into a similar issue as MetaMask? https://medium.com/metamask/incident-postmortem-slow-confirmation-popups-23108bc954bc

Reply:
Added in newest commit 👍